service/Core/Handlers/GenerateEmbeddingsHandler.cs

```diff
@@ -74,6 +74,12 @@ public sealed class GenerateEmbeddingsHandler : IPipelineStepHandler
         this._log.LogDebug("Generating embeddings, pipeline '{0}/{1}'", pipeline.Index, pipeline.DocumentId);
 
         var partitionsFound = false;
+
+        //we have two different strategies for generate embedding, because we need to support bulk embedding
+        //generation where we optimize the embedding generation greatly reducing the number of call to the model.
+        //we will calculate all embedding in a dictionary where the text is the key and the embedding is the value.
+        var embeddingCache = await EmbeddingGeneratorHelper.GetEmbeddingsAsync(pipeline, this, this._embeddingGenerators, this._orchestrator, cancellationToken).ConfigureAwait(false);
```

Comment:
This approach needs to be revisited because the embedding data is lost in case of storage errors below. Not sure yet, I need to think about it before moving forward, I might merge the PR and fix it separately

Reply:
I'm almost done updating the code, will push later after I finish testing